Add a developer page describing the packaging strategy for mantid

martyngigg opened the pull request:
**Description of work.**

It is oriented towards developers to give an overview of where the code is and how the packaging fits into the CI pipeline. It also offers instructions on building locally.


**To test:**

Read the document for clarity.

Fixes #35200

*This does not require release notes* because **this is an internal document**

<!-- Ensure the base of this PR is correct (e.g. release-next or main)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

martyngigg committed:
Add a page describing the different packages mantid produces

It is oriented towards developers to give an overview
of where the code is and how the packaging fits into the CI
pipeline. It also offers instructions on building locally.

thomashampson commented on dev-docs/source/Packaging.rst:
```diff
@@ -0,0 +1,124 @@
+exception of ``mantiddocs`` where a single, ``noarch`` package is created for all
+operating systems as it only contains static HTML.
+
+All mantid-conda packages are available to users through the Mantid's
```
```suggestion
All mantid-conda packages are available to users through Mantid's
```

thomashampson commented on dev-docs/source/Packaging.rst:
```diff
@@ -0,0 +1,124 @@
+operating systems as it only contains static HTML.
+
+All mantid-conda packages are available to users through the Mantid's
+`Conda organization <mantid-conda-org>`_.
```
Missing underscore
```suggestion
`Conda organization <mantid-conda-org_>`_.
```

thomashampson commented on dev-docs/source/Packaging.rst:
```diff
@@ -0,0 +1,124 @@
+Conda
+-----
+
+Mantid provides packages to be consumend by the `Conda <conda_>`_ package manager.
+This is separate to the ``mantid-developer`` enviroment files that setup a local
+development enviroment. See :ref:`GettingStarted` for more detail on how to setup a development enviroment.
```
```suggestion
Mantid provides packages to be consumed by the `Conda <conda_>`_ package manager.
This is separate to the ``mantid-developer`` environment files that setup a local
development environment. See :ref:`GettingStarted` for more detail on how to setup a development environment.
```

thomashampson commented on dev-docs/source/Packaging.rst:
```diff
@@ -0,0 +1,124 @@
+   # wait quite a while ...
+   # packages will appear in a conda-bld directory
+
+To create a new test enviroment with packages from the local build:
```
```suggestion
To create a new test environment with packages from the local build:
```

thomashampson commented on dev-docs/source/Packaging.rst:
```diff
@@ -0,0 +1,124 @@
+`script <package-standalone_>`_ to encapsulate the build steps.
+
+To build an installer locally first build a set of conda packages using the
+`instructions above <building-conda-packages_>`_. Once you have the ``conda-bld``
```
internal reference not working for me
```suggestion
:ref:`instructions above <building-conda-packages>`. Once you have the ``conda-bld``
```
which also needs this above the `Building Conda Packages` header:
```
.. _building-conda-packages:

```

That works for me, but there's probably another way to do it.

thomashampson reviewed (changes requested):
Nice addition to our dev docs. Just a few minor typos and a couple of links that didn't work for me when I built it locally.

martyngigg committed:
Remove ancient Sphinx version checks.

These versions are no longer used in the wild.

martyngigg committed:
Switch from imgmath to mathjax for equations in dev docs

This avoids the build machine needing a latex installation
and matches the online configuration for the user documentation.
If the

thomashampson reviewed (approved):
Thanks for making those changes.
